_docs/policies/tests.md

```diff
@@ -0,0 +1,47 @@
+# `status-go` Test Policy
+
+- [Creating Tests](#creating-tests)
+- [Flaky Tests](#flaky-tests)
+
+## Creating Tests
+
+- All new functionality MUST be introduced with tests that:
+  - Prove that the functionality performs as described
+  - Can be falsified
+  - Are resistant to fuzzing
+- All new tests MUST BE validated via 1000 local tests. Ensuring that the test runs and passes consistently every time gives confidence that the test is not flaky.
```

Comment:
This sentence I sort of disagree. Sort of because perhaps we should differentiate the type of test we want to rerun hundreds of times because a good portion of _unit tests_ are so deterministic in nature that this recommendation seems out of place or wasted effort to even worry about. Could we change the wording to something more flexible or realistic?

The tricky thing is that we haven't ever (or have we?) agreed on how to classify status-go's tests. If they all go in the same bucket as "tests" it becomes harder for us to express which ones we really want devs to increase `-count`.

Reply:
I guess we should say that integration tests should be rerun

Reply:
That is reasonable. Let's change the language to:

```suggestion
- All new integration tests MUST BE validated via at least 10000 local tests. Ensuring that the test runs and passes consistently every time gives confidence that the test is not flaky.
```

I also agree with you @ilmotta we should have categories of test. Maybe we can open an issue and place a link to it in the Go Council page to mark this work as needing consensus.

Reply:
I'm also happy to add a definition of "integration tests."

Reply:
This is done https://github.com/status-im/status-go/issues/4935